tests/sklearn/test_sklearn_models.py

```diff
@@ -2032,3 +2033,63 @@ def test_error_raise_unsupported_pandas_values(model_class, bad_value, expected_
 
     with pytest.raises(ValueError, match=expected_error):
         model.fit(x_train, y_train)
+
+
+@pytest.mark.parametrize(
+    "model_class, parameters",
+    get_sklearn_linear_models_and_datasets()
+    + get_sklearn_tree_models_and_datasets()
+    + get_sklearn_neighbors_models_and_datasets(),
+)
+def test_initialization_variables_match(model_class, parameters, load_data, is_weekly_option):
+    """Test that CML models can be initialized with the same parameters scikit-learn models."""
+    n_bits = N_BITS_THRESHOLD_FOR_SKLEARN_CORRECTNESS_TESTS
+
+    x, y = get_dataset(model_class, parameters, n_bits, load_data, is_weekly_option)
+
+    # Instantiate the model
+    model = instantiate_model_generic(model_class, n_bits=n_bits)
+
+    # Fit the model to create the equivalent sklearn model
+    with warnings.catch_warnings():
+        # Sometimes, we miss convergence, which is not a problem for our test
+        warnings.simplefilter("ignore", category=ConvergenceWarning)
+        model.fit(x, y)
+
+    # Assert the sklearn model has been created
+    assert hasattr(model, "sklearn_model"), "Sklearn model not found"
+
+    # Get the constructor parameters of both the custom and sklearn models
+    cml_params = set(inspect.signature(model.__class__).parameters.keys())
+
+    # Accumulate parameters from all base classes of the sklearn model
+    def get_params(cls):
+        params = set(inspect.signature(cls).parameters.keys())
+        for base in cls.__bases__:
+            params.update(get_params(base))
+        return params
+
+    # Conditionally gather parameters from base classes for XGBClassifier and XGBRegressor
+    if model.sklearn_model.__class__.__name__ in ["XGBClassifier", "XGBRegressor"]:
```

Comment:
why are they  handled differently ?

Reply:
xgboost sklearn api doesn't really follow sklearn on parameter initialization. They have them in the parent class instead.